src/main/java/gift/service/OrderService.java

```diff
@@ -0,0 +1,42 @@
+package gift.service;
+
+import gift.dto.request.OrderRequestDto;
+import gift.entity.Member;
+import gift.entity.Order;
+import gift.entity.ProductOption;
+import gift.repository.MemberRepository;
+import gift.repository.OrderRepository;
+import gift.repository.ProductOptionRepository;
+import gift.repository.WishRepository;
+import jakarta.transaction.Transactional;
+import org.springframework.stereotype.Service;
+
+import java.util.NoSuchElementException;
+
+
+@Service
+public class OrderService {
+
+    private final OrderRepository orderRepository;
+    private final WishRepository wishRepository;
+    private final ProductOptionRepository optionRepository;
+
+    public OrderService(OrderRepository orderRepository, ProductOptionRepository optionRepository, WishRepository wishRepository) {
+        this.orderRepository = orderRepository;
+        this.optionRepository =optionRepository;
+        this.wishRepository = wishRepository;
+    }
+
+    @Transactional
+    public Order placeOrder(Member member, OrderRequestDto dto) {
+        ProductOption option = optionRepository.findById(dto.getOptionId())
+                .orElseThrow(() -> new NoSuchElementException("옵션 ID " + dto.getOptionId() + "가 존재하지 않습니다."));
+        option.subtractQuantity(dto.getQuantity());
+
+        wishRepository.deleteByMemberIdAndProductId(member.getId(), option.getProduct().getId());
+        Order order =new Order(member,option,dto.getQuantity(), dto.getMessage());
```

Comment:
이 부분은 member 객체에게 주문하라(= order)라고 메시지를 보내볼 수도 있을 것 같네요. 😃

```suggestion
        Order order = member.order(option, dto.getQuantity(), dto.getMessage());
```